qiskit_nature/properties/second_quantization/electronic/electronic_driver_result.py

```diff
@@ -0,0 +1,103 @@
+# This code is part of Qiskit.
+#
+# (C) Copyright IBM 2021.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The ElectronicDriverResult class."""
+
+from typing import List, Tuple, cast
+
+from qiskit_nature.drivers import Molecule
+from qiskit_nature.drivers.second_quantization import QMolecule
+from qiskit_nature.operators.second_quantization import FermionicOp
+
+from ..driver_result import DriverResult
+from ..second_quantized_property import LegacyDriverResult, LegacyElectronicDriverResult
+from .angular_momentum import AngularMomentum
+from .bases import ElectronicBasis, ElectronicBasisTransform
+from .dipole_moment import TotalDipoleMoment
+from .electronic_energy import ElectronicEnergy
+from .magnetization import Magnetization
+from .particle_number import ParticleNumber
+
+
+class ElectronicDriverResult(DriverResult):
+    """The ElectronicDriverResult class.
+
+    This is a :class:~qiskit_nature.properties.CompositeProperty gathering all property objects
+    previously stored in Qiskit Nature's `QMolecule` object.
+    """
+
+    def __init__(self) -> None:
+        """
+        Property objects should be added via `add_property` rather than via the initializer.
+        """
+        super().__init__(self.__class__.__name__)
+        self.molecule: Molecule = None
+        self.electronic_basis_transform: ElectronicBasisTransform = None
+        # TODO: add origin driver metadata
+        # TODO: where to put orbital_energies?
+        # TODO: where to put kinetic, overlap matrices? Do we want explicit Fock matrix?
```

Comment:
Finally, we have these additional matrices:
- `kinetic`
- `overlap`
- and potentially for the future of DFT Embedding we will also need a place for `fock` (because the DFT-XC part cannot be constructed from the `eri` alone)

I'd also argue to put these into `ElectronicEnergy` but would like some input first.

Reply:
@pbark @woodsp-ibm We also still need to make a decision about this point

Reply:
Again here these properties are most naturally place in `ElectronicEnergy`. (same arguments as before)